app/grandchallenge/reader_studies/views.py

```diff
@@ -1374,89 +1371,85 @@ def get_success_url(self):
         )
 
 
-class DisplaySetInterfacesCreate(ObjectPermissionRequiredMixin, TemplateView):
-    form_class = SingleCIVForm
-    permission_required = (
-        f"{ReaderStudy._meta.app_label}.change_{DisplaySet._meta.model_name}"
-    )
-    raise_exception = True
-    template_name = "reader_studies/display_set_new_interface_create.html"
-
-    def get_permission_object(self):
-        return self.display_set
-
+class DisplaySetInterfacesCreate(InterfacesCreateBaseView):
     @property
-    def display_set(self):
+    def object(self):
         if self.kwargs.get("pk"):
             return DisplaySet.objects.get(pk=self.kwargs["pk"])
-
-    @property
-    def reader_study(self):
-        if self.display_set:
-            return self.display_set.reader_study
         else:
-            return ReaderStudy.objects.get(slug=self.kwargs["slug"])
+            return None
 
-    def get_form_kwargs(self):
-        return {
-            "pk": self.kwargs.get("pk"),
-            "base_obj": self.reader_study,
-            "interface": self.request.GET.get("interface"),
-            "user": self.request.user,
-            "auto_id": f"id-{uuid.uuid4()}",
-            "htmx_url": self.get_htmx_url(),
-        }
+    @property
+    def base_object(self):
+        return ReaderStudy.objects.get(slug=self.kwargs["slug"])
 
     def get_htmx_url(self):
         if self.kwargs.get("pk") is not None:
             return reverse_lazy(
                 "reader-studies:display-set-interfaces-create",
                 kwargs={
                     "pk": self.kwargs.get("pk"),
-                    "slug": self.reader_study.slug,
+                    "slug": self.base_object.slug,
                 },
             )
         else:
             return reverse_lazy(
                 "reader-studies:display-set-new-interfaces-create",
-                kwargs={"slug": self.reader_study.slug},
+                kwargs={"slug": self.base_object.slug},
             )
 
-    def get_context_data(self, *args, **kwargs):
-        context = super().get_context_data(*args, **kwargs)
-        context.update(
-            {
-                "object": self.display_set,
-                "form": self.form_class(**self.get_form_kwargs()),
-            }
-        )
-        return context
-
 
 class DisplaySetCreateView(
-    BaseAddObjectToReaderStudyMixin,
-    InterfaceProcessingMixin,
-    FormView,
+    ObjectCreateMixin,
+    MultipleCIVProcessingBaseView,
 ):
     form_class = DisplaySetCreateForm
-    template_name = "reader_studies/display_set_create.html"
-    type_to_add = "Display Set"
     permission_required = (
         f"{ReaderStudy._meta.app_label}.change_{ReaderStudy._meta.model_name}"
     )
     included_form_classes = (
         DisplaySetCreateForm,
-        SingleCIVForm,
+        *MultipleCIVProcessingBaseView.included_form_classes,
     )
     success_message = "Display set has been created."
+    type_to_add = DisplaySet._meta.verbose_name
```

Comment:
Better to use the model name rather than the verbose name as the latter can change with the locale.

Reply:
The only downside is that I'm using this in the templates for the breadcrumbs, where the verbose name would be better or else we end up with "Archiveitems" rather than "Archive Items" etc. I can add another variable to the context that contains the verbose name, or do you see another option?

Reply:
Or I add a template tag to retrieve the verbose name of an object. Would avoid cluttering the context of the view. What do you think?

Reply:
ah yeah, but then it might change with locale which we probably don't want. Then maybe `type_to_add` and `base_model_name` shouldn't rely on the object's meta information but be supplied as strings on the view instead.

Reply:
Yes template tags to get the information would be the way to go. `object._meta.model_name` is safe to use as an identifier.